web/src/scripts/actions/storyboardActions.js

```diff
@@ -0,0 +1,182 @@
+ *    it under the terms of the GNU Affero General Public License, version 3,
+ *    as published by the Free Software Foundation.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import _ from 'lodash'
+import path from 'path'
+
+import * as editorActions from '../actions/editorActions'
+import * as storyUtils from '../utils/storyboard'
+import StoryboardChangeFactory from '../factories/storyboard/StoryboardChangeFactory'
+import FileScaffoldFactory from '../factories/scaffold/FileScaffoldFactory'
+import { fileTreeCompositeActions, textEditorCompositeActions } from '../actions'
+
+export const at = {
+  ADD_SCENE: 'ADD_SCENE',
+  DELETE_SCENE: 'DELETE_SCENE',
+  RENAME_SCENE: 'RENAME_SCENE',
+  ADD_CONNECTION: 'ADD_CONNECTION',
+  UPDATE_CONNECTION: 'UPDATE_CONNECTION',
+  DELETE_CONNECTION: 'DELETE_CONNECTION',
+  SET_ENTRY_SCENE: 'SET_ENTRY_SCENE',
+  OPEN_STORYBOARD: 'OPEN_STORYBOARD',
+  TOGGLE_VIEW: 'TOGGLE_VIEW',
+}
+
+export const toggleStoryboardView = () => async (dispatch) => {
+  dispatch({
+    type: at.TOGGLE_VIEW,
+  })
+}
+
+export const openStoryboard = (filepath) => async (dispatch, getState) => {
+  const storyboardDoc = await dispatch(editorActions.getDocument(filepath))
+  await dispatch(editorActions.setCurrentDoc(storyboardDoc.id))
+  const rootPath = getState().directory.rootPath
+  const storyboardCode = storyboardDoc.code
+  const sceneImports = storyUtils.getFilePathsFromStoryboardCode(storyboardCode, {
+    directoryPath: rootPath,
+  })
+  const sceneInfo = storyUtils.getSceneInformationForStoryboardCode(storyboardCode)
+
+  //load in files from imports
+  await Promise.all(_.map(sceneImports, async (sceneImport, filepath) => {
+    await dispatch(editorActions.getDocument(filepath))
+  }))
+
+  //get scenes from storyboard code
+  const docCache = getState().editor.docCache
+
+  let scenes = {...sceneInfo.scenes}
+  //get code from new files
+  //get connections for files
+  const sceneConnections = _.map(sceneImports, ({ sceneName, source }) => {
+    scenes[sceneName].filePath = source
+    storyUtils.buildElementTree(docCache[source].code)
+    return {
+      connections: storyUtils.getConnectionsInCode(docCache[source].code),
+      sceneName,
+    }
+  })
+
+  // match source info to component layout info
+  // return storyboard object
+  dispatch({
+    type: at.OPEN_STORYBOARD,
+    payload: {
+      connections: sceneConnections,
+      scenes: scenes,
+      entry: sceneInfo.entry,
+    },
+  })
+}
+
+export const addScene = () => async (dispatch, getState) => {
+  const scaffoldId = 0 // id corresponds to Component
```

Comment:
Probably shouldn't check for id 0. We can make an API to get by name or something. Or change ids to `com.decosoftware.storyboard` or something, idk